File: action-type/14-stratery/go/main.go
package main

import (
	"math/big"
)

func main() {

	context := &PayContext{}

	// 用户选择支付宝
	context.SetStrategy(&AliPayStrategy{})
	context.Pay(big.NewFloat(100.00))

	// 用户选择微信
	context.SetStrategy(&WeChatPayStrategy{})
	context.Pay(big.NewFloat(200.00))
}

File: action-type/14-stratery/java/Main.java
package main;

import java.math.BigDecimal;

public class Main {
    
    public static void main(String[] args) {

        PayContext context = new PayContext();

        // 用户选择支付宝
        context.setStrategy(new AliPayStrategy());
        context.pay(new BigDecimal("100.00"));

        // 用户选择微信
        context.setStrategy(new WeChatPayStrategy());
        context.pay(new BigDecimal("200.00"));
    }
}
